addons/sale_timesheet/report/project_profitability_report_analysis.py

```diff
@@ -127,8 +127,17 @@ def init(self):
                                 LEFT JOIN sale_order_line_invoice_rel SOINV ON SOINV.invoice_line_id = INVL.id
                                 LEFT JOIN sale_order_line SOL ON SOINV.order_line_id = SOL.id
                                                              AND SOL.product_id = AAL.product_id
```

Comment:
Your fix makes me think again about those lines which could be insufficiently restrictive (we search for a too specific case imho)

If a user creates an invoice independently from his project, uses the project analytic account and then create a credit note on this invoice, we will count the credit note as an expense cost, which is, I think, not expected.
Maybe we can remove those lines
```suggestion
```
 and test :
```diff
AND INVL.id IS NULL -- exclude credit notes from this subquery
```
herebelow.

In this case the query will be cristal clear, imho : We search for AAL which are not a reversal and has not been reversed.

What do you think @IT-Ideas ?

Reply:
LGTM Thanks @tlibioulle 🙂